File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/MifosGroupListAdapter.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.mifosxdroid.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.mifos.mifosxdroid.R;
import com.mifos.objects.db.MifosGroup;

import java.util.List;


public class MifosGroupListAdapter extends BaseAdapter {

    private LayoutInflater layoutInflater;
    private List<MifosGroup> groups;

    public MifosGroupListAdapter(Context context, List<MifosGroup> groups){
        layoutInflater = LayoutInflater.from(context);
        this.groups = groups;
    }

    @Override
    public int getCount() {
        return this.groups.size();
    }

    @Override
    public MifosGroup getItem(int i) {
        return this.groups.get(i);
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {

        ViewHolder viewHolder;

        if(view==null){
            view = layoutInflater.inflate(R.layout.row_group_list_item,viewGroup,false);
            viewHolder = new ViewHolder();
            view.setTag(viewHolder);
        }else{
            viewHolder = (ViewHolder) view.getTag();
        }

        viewHolder.tv_group_name = (TextView) view.findViewById(R.id.tv_group_name);
        viewHolder.tv_staff_name = (TextView) view.findViewById(R.id.tv_staff_name);
        viewHolder.tv_level_name = (TextView) view.findViewById(R.id.tv_level_name);


        MifosGroup mifosGroup = groups.get(i);
        viewHolder.tv_group_name.setText(mifosGroup.getGroupName());
        viewHolder.tv_staff_name.setText( mifosGroup.getStaffName());
        viewHolder.tv_level_name.setText(mifosGroup.getLevelName());

        return view;
    }

    public static class ViewHolder
    {
        TextView tv_group_name;
        TextView tv_staff_name;
        TextView tv_level_name;
    }
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/MifosGroupListAdapter.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.mifosxdroid.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.mifos.core.objects.db.MifosGroup
import com.mifos.mifosxdroid.databinding.RowGroupListItemBinding

class MifosGroupListAdapter(context: Context?, groups: List<MifosGroup>) : BaseAdapter() {
    private val layoutInflater: LayoutInflater
    private val groups: List<MifosGroup>

    init {
        layoutInflater = LayoutInflater.from(context)
        this.groups = groups
    }

    override fun getCount(): Int {
        return groups.size
    }

    override fun getItem(i: Int): MifosGroup {
        return groups[i]
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
        val binding: RowGroupListItemBinding
        val convertView: View
        if (view == null) {
            binding = RowGroupListItemBinding.inflate(layoutInflater, viewGroup, false)
            convertView = binding.root
            convertView.tag = binding
        } else {
            binding = view.tag as RowGroupListItemBinding
            convertView = view
        }
        val mifosGroup = groups[i]
        binding.tvGroupName.text = mifosGroup.groupName
        binding.tvStaffName.text = mifosGroup.staffName
        binding.tvLevelName.text = mifosGroup.levelName
        return convertView
    }
}